src/OpenSage.Game/Graphics/ParticleSystems/ParticleSystem.cs

```diff
@@ -360,29 +376,12 @@ private void UpdateParticle(ref Particle particle)
             particle.AngleZ += particle.AngularRateZ;
             particle.AngularRateZ *= particle.AngularDamping;
 
-            ParticleColorKeyframe nextC = null;
-            ParticleColorKeyframe prevC = null;
-            foreach (var colorKeyframe in _colorKeyframes)
-            {
-                if (colorKeyframe.Time >= particle.Timer)
-                {
-                    nextC = colorKeyframe;
-                    break;
-                }
-                prevC = colorKeyframe;
-            }
-            if (prevC == null)
-            {
-                prevC = _colorKeyframes[0];
-            }
-            if (nextC == null)
-            {
-                nextC = prevC;
-            }
-            if (prevC != nextC)
+            FindKeyframes(particle.Timer, _colorKeyframes, out var nextC, out var prevC);
+
+            if (!prevC.Equals(nextC))
             {
                 var colorInterpoland = (float) (particle.Timer - prevC.Time) / (nextC.Time - prevC.Time);
-                particle.Color = Vector3Utility.Lerp(ref prevC.Color, ref nextC.Color, colorInterpoland);
+                particle.Color = Vector3Utility.Lerp(in prevC.Color, in nextC.Color, colorInterpoland);
```

Comment:
I haven't been using the `in` keyword in call sites - it's optional, and I think not including it better matches the default behaviour for reference types.